lib/game.rb

```diff
@@ -0,0 +1,68 @@
+class Game
+	attr_accessor :board, :player_1, :player_2
+
+	WIN_COMBINATIONS = [
+		[0,1,2],
+		[3,4,5],
+		[6,7,8],
+		[0,3,6],
+		[1,4,7],
+		[2,5,8],
+		[0,4,8],
+		[6,4,2]
+	]
+
+	def initialize
+	end
+
+	def current_player(board)
+		@board.turn_count % 2 == 0 ? @player_1 : @player_2
+	end
+
+	def won?
+		WIN_COMBINATIONS.detect do |combo|
+			@board.cells[combo[0]] == @board.cells[combo[1]] &&
+			@board.cells[combo[1]] == @board.cells[combo[2]] &&
+			@board.taken?(combo[0]+1)
+	end
+
+	def draw?
+		@board.full? && !won?
+	end
+
+	def over?
+		won? || draw?
+	end
+
+	def winner
+		if winning_combo = won?
+			@winner = @board.cells[winning_combo.first]
+		end
+	end
+
+	def play
+		while !over?
+	      turn
+	    end
+	    if won?
+	      puts "Congratulations #{winner}!"
+	    elsif draw?
+	      puts "Cats Game!"
+	    end
+	end
+
+	def turn
+	  player = current_player
+      current_move = player.move(@board)
+    if !@board.valid_move?(current_move)
+  	  turn
+    else
+      puts "Turn: #{@board.turn_count+1}\n"
```

Comment:
We would prefer to see the CLI logic separated from the game logic in its own class (e.g. TicTacToeCLI). Splitting the interface from the implementation of the game allows for multiple interfaces (e.g. a web app) without requiring changes to the game code.